posthog/temporal/batch_exports/postgres_batch_export.py

```diff
@@ -133,21 +133,34 @@ async def connect(
             # Disable certificate verification for self-signed certificates.
             kwargs["sslrootcert"] = None
 
+        max_attempts = 5
         connect = make_retryable_with_exponential_backoff(
             psycopg.AsyncConnection.connect,
-            retryable_exceptions=(psycopg.OperationalError,),
+            max_attempts=max_attempts,
+            retryable_exceptions=(psycopg.OperationalError, psycopg.errors.ConnectionTimeout),
         )
 
-        connection: psycopg.AsyncConnection = await connect(
-            user=self.user,
-            password=self.password,
-            dbname=self.database,
-            host=self.host,
-            port=self.port,
-            connect_timeout=self.connection_timeout,
-            sslmode="prefer" if settings.TEST else "require",
-            **kwargs,
-        )
+        try:
+            connection: psycopg.AsyncConnection = await connect(
+                user=self.user,
+                password=self.password,
+                dbname=self.database,
+                host=self.host,
+                port=self.port,
+                connect_timeout=self.connection_timeout,
+                sslmode="prefer" if settings.TEST else "require",
+                **kwargs,
+            )
+        except psycopg.errors.ConnectionTimeout as err:
+            raise PostgreSQLConnectionError(
+                f"Timed-out while trying to connect for {max_attempts} attempts. Is the "
+                f"server running at '{self.host}', port '{self.port}' and accepting "
+                "TCP/IP connections?"
+            ) from err
+        except psycopg.OperationalError as err:
+            raise PostgreSQLConnectionError(
+                f"Failed to connect after {max_attempts} attempts. Please review connection configuration."
+            ) from err
```

Comment:
Moving forward, we should start doing more of this: Raising our own exceptions with more precise details of what could have happened, as these are relayed to the user via logs.

In this case, I can see two things going wrong:
* psycopg.errors.ConnectionTimeout: If the database is unreachable.
* psycopg.OperationalError: If the credentials or other connection details are wrong, but the database is reachable.